### ISSUE TITLE
Invalid Count Argument

### ISSUE THREAD

**Issue by tmanochehri** [TechToSpeech/terraform-aws-serverless-static-wordpress#38]
Hi there! Very new to Terraform. Any advice is appreciated.

Error message:
![Error1](https://user-images.githubusercontent.com/18153898/123535375-56a70b00-d6d8-11eb-9604-3c8e71a0238e.PNG)

Main.tf 1 - Account ID Hidden
![MainTF1](https://user-images.githubusercontent.com/18153898/123535408-72121600-d6d8-11eb-9ec1-32a70ec3859e.PNG)

Main.tf 2
![MainTF2](https://user-images.githubusercontent.com/18153898/123535464-bf8e8300-d6d8-11eb-9675-90806516a7e8.PNG)

VPC:
![VPC](https://user-images.githubusercontent.com/18153898/123535495-f82e5c80-d6d8-11eb-870c-e859a48006ee.PNG)

Subnets:
![subnets](https://user-images.githubusercontent.com/18153898/123535503-11370d80-d6d9-11eb-9709-f76639e1e06a.PNG)

Provider.tf:
![provider](https://user-images.githubusercontent.com/18153898/123535520-2e6bdc00-d6d9-11eb-9dfc-f65b4512f145.PNG)

Thank you! Really enjoying working on this.

**Comment by petewilcock** [comment]
The VPC module is just included as an example for VPC creation if you don't have one already - if you do (and it looks like you do as you've inserted your values there), you can just remove the `module_vpc` block and it'll work as expected :)

**Comment by tmanochehri** [comment]
Thank you Pete, that did work. I don't think I setup the subnets correctly, though (I didn't put them in different availability zones, for example) so I ran into other issues.

I want to start from scratch, so I deleted the manually created VPC, deleted all cached terraform files and re-ran terraform with just the *_VPC module, but I get the same count argument errors that I noted above.

I do have a default VPC in the us-west-1 region, could that be causing a conflict?

**Comment by petewilcock** [comment]
No I think I've actually just written that VPC module quite poorly, The errors are because I'm creating a specific number of resources based on a value that Terraform can't work out in advance. It seems that older versions used to be a bit more tolerant of this but now it just throws a hard error. 

It comes from the fact that easy region has a variable number of availability zones, at least two but sometimes five. What I tried to do there was query the number of zones and then create as many as needed, hence the count resource. Basically I just need to rewrite that with a better method. 

TL;DR though, I'd just create your fresh VPC and availability zones manually and plug the values back in until I fix that up. You only really need one AZ for this but two is good practice if you ever plan to deploy anything else in future :)

**Comment by tmanochehri** [comment]
Okay, thank you for your input! I will get this closed out.
